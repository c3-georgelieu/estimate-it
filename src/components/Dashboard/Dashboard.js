// TODO: implement delete project

import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import moment from 'moment'
import {
  Container,
  Row,
  Col,
  ListGroup,
  ListGroupItem,
  Badge,
  Button,
} from 'reactstrap'
import FA from 'react-fontawesome'
import Header from '../Header'
import { estimateType } from '../Estimate/propTypes'

const Dashboard = ({ estimates }) => (
  <div className="dashboard">
    <Header />
    <Container>
      <Row>
        <Col lg={6}>
          <ListGroup>
            {Object.values(estimates).map(({
              project,
              _id,
              _changed,
              modifiedBy,
            }) => (
              // TODO: move to a separate component
              <ListGroupItem key={_id}>
                <Row>
                  <Col xs={9} sm={10} lg={10}>
                    <Row>
                      <Col xs={12} sm={6} lg={12}>
                        <Link to={`/estimate/${_id}`}>
                          <b>
                            {(_id === 'new' && 'New Project') || project || _id}
                          </b>
                        </Link>
                      </Col>
                      {_id !== 'new' && (
                        <Col xs={12} sm={6} lg={12}>
                          <Badge pill>
                            {moment(_changed).fromNow()}
                            {' by '}
                            {modifiedBy}
                          </Badge>
                        </Col>
                      )}
                    </Row>
                  </Col>
                  {_id !== 'new' && (
                    <Col xs={3} sm={2} lg={2}>
                      <Button outline color="danger">
                        <FA name="trash" />
                      </Button>
                    </Col>
                  )}
                </Row>
              </ListGroupItem>
            ))}
          </ListGroup>
        </Col>
        <Col>
          Other stuffâ€¦
        </Col>
      </Row>
    </Container>
  </div>
)

Dashboard.propTypes = {
  estimates: PropTypes.objectOf(estimateType).isRequired,
}

export default Dashboard
