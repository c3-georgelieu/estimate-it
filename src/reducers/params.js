import { SET_PARAMS } from '../actions/types'

const defaultState = {
  apiKey: '',
  estimateId: '',
}

export default (state = defaultState, { type, payload }) => {
  switch (type) {
    case SET_PARAMS:
      return ({
        ...state,
        apiKey: payload.apiKey,
        estimateId: payload.estimateId,
      })

    default:
      return state
  }
}
