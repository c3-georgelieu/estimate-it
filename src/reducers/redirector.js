import { REDIRECT } from '../actions/types'

const defaultState = {
  pathToRedirect: null,
}

export default (state = defaultState, { type, payload }) => {
  switch (type) {
    case REDIRECT:
    console.log(state)
      return ({
        pathToRedirect: payload.pathToRedirect,
      })

    default:
      return state
  }
}
