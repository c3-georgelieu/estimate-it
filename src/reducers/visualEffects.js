import { TOGGLE_SPINNER } from '../actions/types'

const defaultState = {
  showSpinner: false,
}

export default (state = defaultState, { type, payload }) => {
  switch (type) {
    case TOGGLE_SPINNER:
      return ({
        showSpinner: payload.showSpinner,
      })

    default:
      return state
  }
}
