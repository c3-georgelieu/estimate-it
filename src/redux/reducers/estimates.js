import { handleText } from '../../helpers/task-parsing'
import {
  ADD_ESTIMATE,
  RECALCULATE,
  CLEAN_ESTIMATE,
} from '../actions/types'

const emptyEstimate = ({
  _id: 'new',
  text: '',
  graphData: [],
  project: 'New Project',
})

const defaultState = {
  new: emptyEstimate,
}

export default (state = defaultState, { type, payload }) => {
  switch (type) {
    case ADD_ESTIMATE:
      return ({
        ...state,
        [payload._id]: { ...payload },
      })
    case CLEAN_ESTIMATE:
      return ({
        ...state,
        // Put 'new' here to avoid creating an estimate with an undefined `_id`
        // ...while logging out on different routes
        // ...(that may not contain an `:estimateId` param)
        [payload.estimateId || 'new']: emptyEstimate,
      })
    case RECALCULATE:
      return ({
        ...state,
        [payload._id]: {
          _id: payload._id,
          ...handleText(state[payload._id].text),
        },
      })

    default:
      return state
  }
}
