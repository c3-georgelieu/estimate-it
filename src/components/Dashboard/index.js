import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import Dashboard from './Dashboard'
import extract from '../../helpers/extract'

const mapStateToProps = extract({ estimates: 1 })

const mapDispatchToProps = ({})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Dashboard))
