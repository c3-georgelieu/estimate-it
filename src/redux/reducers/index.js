import { combineReducers } from 'redux'
import { reducer as toastrReducer } from 'react-redux-toastr'
import estimates from './estimates'
import visualEffects from './visualEffects'
import redirector from './redirector'
import creds from './creds'

export default combineReducers({
  estimates,
  visualEffects,
  toastr: toastrReducer,
  redirector,
  creds,
})
