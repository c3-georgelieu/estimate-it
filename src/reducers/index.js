import { combineReducers } from 'redux'
import estimate from './estimate'

export default combineReducers({
  estimate,
})
