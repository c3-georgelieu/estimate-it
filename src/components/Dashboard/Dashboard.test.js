/* globals describe, it, expect */

import React from 'react'
import renderer from 'react-test-renderer'
import { HashRouter as Router } from 'react-router-dom'
import Dashboard from './Dashboard'
import { emptyEstimate } from '../../redux/reducers/estimates'

describe('Dashboard', () => {
  it('renders defaults correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Dashboard
            estimates={{ emptyEstimate }}
          />
        </Router> // eslint-disable-line comma-dangle
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
