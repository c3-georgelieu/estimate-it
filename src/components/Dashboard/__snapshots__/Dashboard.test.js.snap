// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders defaults correctly 1`] = `
<div
  className="dashboard"
>
  <nav
    className="navbar navbar-expand navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Estimate It!
    </a>
    <ul
      className="ml-auto navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="https://github.com/tatomyr/estimate-it"
          onClick={[Function]}
        >
          GitHub
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-6"
      >
        <ul
          className="list-group"
        >
          <li
            className="list-group-item"
          >
            <div
              className="row"
            >
              <div
                className="col-9 col-sm-10 col-lg-10"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-12 col-sm-6 col-lg-12"
                  >
                    <a
                      href="#/estimate/new"
                      onClick={[Function]}
                    >
                      <b>
                        New Project
                      </b>
                    </a>
                  </div>
                  <div
                    className="col-12 col-sm-6 col-lg-12"
                  />
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
      <div
        className="col"
      >
        Other stuffâ€¦
      </div>
    </div>
  </div>
</div>
`;
